src/needless_bool.rs

```diff
@@ -78,6 +93,69 @@ impl LateLintPass for NeedlessBool {
     }
 }
 
+#[derive(Copy,Clone)]
+pub struct BoolComparison;
+
+impl LintPass for BoolComparison {
+    fn get_lints(&self) -> LintArray {
+        lint_array!(BOOL_COMPARISON)
+    }
+}
+
+impl LateLintPass for BoolComparison {
+    fn check_expr(&mut self, cx: &LateContext, e: &Expr) {
+        if let ExprBinary(Spanned{ node: BiEq, .. }, ref left_side, ref right_side) = e.node {
+            match (fetch_bool_expr(left_side), fetch_bool_expr(right_side)) {
+                (Some(true), None) => {
+                    let side_snip = snippet(cx, right_side.span, "..");
+                    let hint = format!("`{}`", side_snip);
+                    span_lint_and_then(cx,
+                                       BOOL_COMPARISON,
+                                       e.span,
+                                       "equality checks against booleans are unnecesary",
```

Comment:
Actually, since we know which boolean it is, perhaps we could make this clearer?

"equality checks against `true` are unnecessary"

and 

"equality checks against `false` can be replaced by a negation"

for the two cases